plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java

```diff
@@ -135,6 +137,8 @@ public static Connector createConnector(
             Set<ConnectorTableFunction> connectorTableFunctions = injector.getInstance(Key.get(new TypeLiteral<Set<ConnectorTableFunction>>() {}));
             FunctionProvider functionProvider = injector.getInstance(FunctionProvider.class);
 
+            checkState(!injector.getBindings().containsKey(Key.get(HiveConfig.class)), "HiveConfig should not be bound");
```

Comment:
nit: this look over-specific - after all we are not checking other class which we do not expect to be bound if those are not.

Reply:
HiveConfig is just the most likely offender (based on history of delta connector), but i am open to suggestoins